source/MaterialXTest/ShaderValid.cpp

```diff
@@ -0,0 +1,291 @@
+#include <MaterialXTest/Catch/catch.hpp>
+
+#include <MaterialXCore/Document.h>
+#include <MaterialXFormat/XmlIo.h>
+
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/GlslShaderGenerator.h>
+#include <MaterialXShaderGen/Util.h>
+#include <MaterialXShaderGen/ShaderGenerators/Common/Swizzle.h>
+#include <MaterialXShaderGen/HwShader.h>
+
+#include <fstream>
+
+namespace mx = MaterialX;
+
+// Compile if module flags were set
+#if defined(MATERIALX_TEST_VIEW) && defined(MATERIALX_BUILD_VIEW)
+// Run only on windows for now
+#if defined(_WIN32)
+#include <iostream>
+#include <MaterialXView/ShaderValidators/Glsl/GlslValidator.h>
+#include <MaterialXView/Glsl/external/tinyexr/TinyEXRImageHandler.h>
+
+TEST_CASE("GLSL Validation from Source", "[view]")
+{
+    // Initialize a GLSL validator. Will initialize 
+    // window and context as well for usage
+    mx::GlslValidator validator;
+    bool initialized = false;
+    try
+    {
+        validator.initialize();
+        initialized = true;
+    }
+    catch (mx::ExceptionShaderValidationError e)
+    {
+        for (auto error : e._errorLog)
+        {
+            std::cout << e.what() << " " << error << std::endl;
+        }
+    }
+    REQUIRE(initialized);
+
+    std::vector<std::string> shaderNames =
+    {
+        "conditional_test1",
+        "simple_test1_graphoutput",
+        "simple_test1_node",
+        "geometric_nodes",
+        "subgraph_ex1",
+        //"subgraph_ex2" -- cannot validate as lighting information isn't complete yet.
+    };
```

Comment:
This test assumes the shader files has already been generated by other tests, so it will fail if the other tests has not been run previously, or if the other tests changes. Would be better to make it self-contained and only run test on shader code generated from within.

Reply:
I will look into this for now I will make it so that it won't fail if the files can't be found.